docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -852,11 +852,35 @@ In addition, custom converters may be registered by adding their fully qualified
 
 Though these implicit converters use reflection, Quarkus will automatically ensure that they are loaded at the appropriate time.
 
+===== Optional Values
+
+If the configuration type is one of the optional types, then empty values are allowed for the configuration key; otherwise,
+specification of an empty value will result in a configuration error which prevents the application from starting.
```

Comment:
I think we should make it clear that it also concerns Lists/Sets.

Btw does it also concern Maps?

Reply:
No, Maps are handled separately from the conversion infrastructure.  I can update the doc to mention collection types.